src/erlfmt.erl

```diff
@@ -214,6 +218,52 @@ replace_pragma_comment_block(_Prefix, [("%" ++ _) = Head | Tail]) ->
 replace_pragma_comment_block(Prefix, [Head | Tail]) ->
     [Head | replace_pragma_comment_block(Prefix, Tail)].
 
+% This 'enclosing range' variants don't expect you to provide
```

Comment:
I like the comment and maybe this is what we want the behaviour of format_file_range to be and so we can think about moving the comment there, but I don't think we should have two different range formatting functions.  I hope that makes sense.

Reply:
I agree that both the API and the name of `format_file_range` are a bit weird. But,
* I'd rather keep some kind of separation (finding the enclosing range VS formatting this range). We **could** refactor to better split those concerns, but I'd rather do that at the end, when more tests are in place, and scope and design are set.
* I think the entry point (at least for the CLI) should give back the whole file, rather that just the modified part. Next commit is already rearranging those functions. We'll be able to discuss better naming then.

Reply:
So it is a bit hard for me to imagine what is to come that will justify having a larger API.  I don't think I understand in what context an IDE would support two different modes of formatting in ranges?

Reply:
* For IDE, i think it expects only the formatted extract, but I have to double-check that.
* For CLI, it's more convenient for the user to return the whole file, even if only a range was formatted. That also will be internally useful for the 'write in-place' variant. 

Then, I prefer small dedicated functions, which might not be part of the public API. By the way I think we should separate public vs internal functions (but needed for unit test) in `-export` list.

Reply:
@michalmuskala Why do we have the `possible_ranges ` function?
My suggestion is that we pick a policy, for example like @slaykachu has done here:
```
            {Starts, Ends} = lists:unzip(PossibleRanges),
            Start = lists:min(Starts),
            End = lists:max(Ends),
```
Then `format_range` always returns `{ok, NodesInRange}` or `{error, Error}` and does not need to return `{options, PossibleRanges}`
What do you guys think?

I am basing this on the suspicion that the Language Server Protocol won't have a need for options and only needs some formatting to happen.

Reply:
FYI @pergu answered here: https://github.com/WhatsApp/erlfmt/pull/6#discussion_r413847386

That's why we shouldn't answer to questions in the review discussions, but in the code itself!
Also, the 'why' is always useful to add (in comments and in commit description).

Reply:
Yes there are multiple options to pick from given a range.  I think we should just pick one to start with.
I am happy with `{1,2,3}` or `{2}`

Reply:
That's exactly what this 'higher lever' function is doing: picking the biggest range, so that every touched form is eventually formatted.
Should I change the function description to make it clearer? I'm not sure how though! :)

Reply:
No I think the description is good.
I think we should replace the implementation of `format_file_range` with `format_file_enclosing_range`.